lib/pos_snapshot_entries.go

```diff
@@ -207,6 +214,24 @@ func (bav *UtxoView) SnapshotCurrentValidators(snapshotAtEpochNumber uint64) err
 		if !validatorEntry.isDeleted {
 			// We only want to snapshot !isDeleted ValidatorEntries.
 			bav._setSnapshotValidatorEntry(validatorEntry, snapshotAtEpochNumber)
+
```

Comment:
should we flip this around such that we pull in everything from the db and merge it into the view and then do one loop over the view?

Reply:
IMO no since this is actually the more efficient operation. If we pull everything from the db, we are loading the ValidatorEntries already in the UtxoView into memory a second time and have a small overhead to loop through those ValidatorEntries and just skip them since they're already in the UtxoView (but that's more negligible than just the double-loading them into memory issue IMO). But LMK if you think the simplicity of the other method outweighs the efficiency gain.

Reply:
nah keep it.